homeassistant/components/homematicip_cloud/sensor.py

```diff
@@ -396,6 +435,143 @@ def native_value(self) -> float:
         return self._device.temperatureExternalDelta
 
 
+@dataclass(kw_only=True, frozen=True)
+class HmipEsiSensorEntityDescription(SensorEntityDescription):
+    """SensorEntityDescription for HmIP Sensors."""
+
+    value_fn: Callable[[AsyncEnergySensorsInterface], StateType]
+    exists_fn: Callable[[FunctionalChannel], bool]
+    type_fn: Callable[[AsyncEnergySensorsInterface], str]
+
+
+SENSORS_ESI = {
+    ESI_CONNECTED_SENSOR_TYPE_IEC: [
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_CURRENT_POWER_CONSUMPTION,
+            native_unit_of_measurement=UnitOfPower.WATT,
+            device_class=SensorDeviceClass.POWER,
+            state_class=SensorStateClass.MEASUREMENT,
+            value_fn=lambda device: device.functional_channel.currentPowerConsumption,
+            exists_fn=lambda channel: channel.currentPowerConsumption is not None,
+            type_fn=lambda device: "CurrentPowerConsumption",
+        ),
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_ENERGY_COUNTER_USAGE_HIGH_TARIFF,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+            value_fn=lambda device: device.functional_channel.energyCounterOne,
+            exists_fn=lambda channel: channel.energyCounterOneType != ESI_TYPE_UNKNOWN,
+            type_fn=lambda device: device.functional_channel.energyCounterOneType,
+        ),
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_ENERGY_COUNTER_USAGE_LOW_TARIFF,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+            value_fn=lambda device: device.functional_channel.energyCounterTwo,
+            exists_fn=lambda channel: channel.energyCounterTwoType != ESI_TYPE_UNKNOWN,
+            type_fn=lambda device: device.functional_channel.energyCounterTwoType,
+        ),
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_ENERGY_COUNTER_INPUT_SINGLE_TARIFF,
```

Comment:
This is energy returned to the grid, but with the LED-impulse version you're using the same key for its lone usage channel. Not sure if this is as intended.

Reply:
I have replaced it with the ESI_TYPE_ENERGY_COUNTER_USAGE_HIGH_TARIFF type.